web/src/main/java/org/hillview/RpcObjectManager.java

```diff
@@ -162,10 +167,10 @@ private void rebuild(RpcTarget.Id id, Observer<RpcTarget> toNotify) {
         HillviewComputation computation = this.generator.get(id);
         if (computation != null) {
             // The following may trigger a recursive reconstruction.
-            HillviewLogger.instance.info("Replaying", "{0}", computation);
+            HillviewLogger.instance.info("Replaying", "computation={0}", computation);
             computation.replay(toNotify);
         } else {
-            Exception ex = new RuntimeException("Cannot reconstruct " + id);
+            Exception ex = new RuntimeException("Don't know how to reconstruct " + id);
```

Comment:
What execution might lead to the 'else' branch?

Reply:
This should not happen, but I've seen many things that should not happen actually happen.

Reply:
Actually, maybe there is a race when many threads remove the same object?
E.g., there is UI to remove objects...

Reply:
Sorry, wrong response to your question. This can happen if a malicious browser client sends a request to access an object with an unknown guid. It can also happen currently because we rebooted the root server and we currently we do not save the redo log on persistent storage.